diff --git a/src/plugins/rpm_extractor.c b/src/plugins/rpm_extractor.c
index 06a817e..66505cb 100644
--- a/src/plugins/rpm_extractor.c
+++ b/src/plugins/rpm_extractor.c
@@ -25,9 +25,12 @@
 #include "platform.h"
 #include "extractor.h"
 #include <stdint.h>
-#include <rpm/rpmlib.h>
-#include <rpm/rpmts.h>
-#include <rpm/rpmlog.h>
+#include <rpmcli.h>
+#include <rpmcli.h>
+#include <rpmio.h>
+#include <rpmtag.h>
+#include <rpmts.h>
+#include <rpmlog.h>
 #include <pthread.h>
 #include <sys/types.h>
 #include <signal.h>
@@ -251,7 +254,7 @@ EXTRACTOR_rpm_extract_method (struct EXTRACTOR_ExtractContext *ec)
   const char *str;
   Header hdr;
   HeaderIterator hi;
-  rpmtd p;
+  HE_t p;
   int i;
   FD_t fdi;
   rpmRC rc;
@@ -310,19 +313,29 @@ EXTRACTOR_rpm_extract_method (struct EXTRACTOR_ExtractContext *ec)
       goto END;
     }
   pthread_mutex_unlock (&parg.lock);
-  hi = headerInitIterator (hdr);
-  p = rpmtdNew ();
-  while (1 == headerNext (hi, p))
+  hi = headerInit(hdr);
+  p = (HE_t)memset(alloca(sizeof(*p)), 0, sizeof(*p));
+  while (1 == headerNext (hi, p, 0))
     for (i = 0; 0 != tests[i].rtype; i++)
       {
 	if (tests[i].rtype != p->tag)
 	  continue;
-	switch (p->type)
+	switch (p->t)
 	  {
+/*rpm5 patch here*/
 	  case RPM_STRING_ARRAY_TYPE:
+	if (0 != proc (proc_cls, 
+				"rpm",
+				tests[i].type,
+				EXTRACTOR_METAFORMAT_UTF8,
+				"text/plain",
+				p->p.str,
+				strlen (p->p.str) +1))
+	return 1;
+	       break;
 	  case RPM_I18NSTRING_TYPE:
 	  case RPM_STRING_TYPE:
-	    while (NULL != (str = rpmtdNextString (p))) 
+	for(p->ix = 0; p->ix < p->c; p->ix++)
 	      {
 		pthread_mutex_lock (&parg.lock);
 		if (0 != ec->proc (ec->cls, 
@@ -330,8 +343,8 @@ EXTRACTOR_rpm_extract_method (struct EXTRACTOR_ExtractContext *ec)
 				   tests[i].type,
 				   EXTRACTOR_METAFORMAT_UTF8,
 				   "text/plain",
-				   str,
-				   strlen (str) + 1))
+				   p->p.argv[p->ix],
+				   strlen (p->p.argv[p->ix]) +1))
 		  
 		  {		 
 		    pthread_mutex_unlock (&parg.lock);
@@ -340,7 +353,7 @@ EXTRACTOR_rpm_extract_method (struct EXTRACTOR_ExtractContext *ec)
 		pthread_mutex_unlock (&parg.lock);
 	      }
 	    break;
-	  case RPM_INT32_TYPE:
+	  case RPM_UINT32_TYPE:
 	    {
 	      if (p->tag == RPMTAG_BUILDTIME)
 		{
